Sources/AsyncHTTPClient/ConnectionPool.swift

```diff
@@ -15,11 +15,16 @@
 enum ConnectionPool {
     /// Used by the `ConnectionPool` to index its `HTTP1ConnectionProvider`s
     ///
-    /// A key is initialized from a `URL`, it uses the components to derive a hashed value
+    /// A key is initialized from a `Request`, it uses the components to derive a hashed value
     /// used by the `providers` dictionary to allow retrieving and creating
     /// connection providers associated to a certain request in constant time.
     struct Key: Hashable, CustomStringConvertible {
+        var scheme: Scheme
+        var connectionTarget: ConnectionTarget
```

Comment:
FYI: For now its probably okay to use `ConnectionTarget` here and it also does not change the behaviour compared to the old implementation. However, I think in the future we should probably use a different type that normalises the `ConnectionTarget`. E.g. the host should be compare case insensitive and different spellings of the same IP address should also result in the same `ConnectionPool.Key`.  Otherwise we end up with two different connections if a user wants to connect to `https://apple.com` and `https://APPLE.com` which we do not want.